configure_frontends.sh

```diff
@@ -14,7 +14,7 @@ function getProperty {
 function setProperty {
   local PROP_KEY=$1
   local PROP_VALUE=$2
-  sed -i '' "s/\(${PROP_KEY}[[:space:]]*=[[:space:]]*\).*\$/\1${PROP_VALUE}/" $GRADLE_PROPERTIES_FILE
+  sed -i "s/\(${PROP_KEY}[[:space:]]*=[[:space:]]*\).*\$/\1${PROP_VALUE}/" $GRADLE_PROPERTIES_FILE
```

Comment:
unfortunately the '' is needed after the -i flag on macOS.
I had a quick talk with chatGPT with suggested following solution
```suggestion
  if [[ "$(uname)" == "Darwin" ]]; then
    sed -i '' "s/\(${PROP_KEY}[[:space:]]*=[[:space:]]*\).*\$/\1${PROP_VALUE}/" $GRADLE_PROPERTIES_FILE
  else
    sed -i "s/\(${PROP_KEY}[[:space:]]*=[[:space:]]*\).*\$/\1${PROP_VALUE}/" $GRADLE_PROPERTIES_FILE
  fi
}
```